src/components/NewConnectionConfig.tsx

```diff
@@ -0,0 +1,247 @@
+import React from 'react';
+import { Input, Select, ButtonGroup, ToolbarButton, Collapse, Field } from '@grafana/ui';
+import {
+  onUpdateDatasourceJsonDataOptionSelect,
+  onUpdateDatasourceResetOption,
+  onUpdateDatasourceJsonDataOption,
+  onUpdateDatasourceSecureJsonDataOption,
+  SelectableValue,
+} from '@grafana/data';
+import { AwsAuthType } from '../types';
+import { awsAuthProviderOptions } from '../providers';
+import { ConfigSection, ConfigSubSection } from '@grafana/experimental';
+import { ConnectionConfigProps } from './ConnectionConfig';
+
+interface NewConnectionConfigProps extends ConnectionConfigProps {
+  currentProvider?: SelectableValue<AwsAuthType> | undefined;
+  awsAllowedAuthProviders: string[];
+  isARNInstructionsOpen: boolean;
+  setIsARNInstructionsOpen: (isOpen: boolean) => void;
+  awsAssumeRoleEnabled: boolean;
+  regions: SelectableValue[];
+  assumeRoleInstructionsStyle: string;
+}
+
+export const NewConnectionConfig = ({
+  isARNInstructionsOpen,
+  setIsARNInstructionsOpen,
+  awsAssumeRoleEnabled,
+  currentProvider,
+  awsAllowedAuthProviders,
+  skipHeader,
+  regions,
+  assumeRoleInstructionsStyle,
+  ...props
+}: NewConnectionConfigProps) => {
+  const options = props.options;
```

Comment:
just curious why we pull options here rather than use the spread up above?

Reply:
We're passing props to the handlers, but we need options in there too. In a few other places in the file we just need options, so I don't necessarily need the spread and can use props.options but  decided to leave the shorthand